### ISSUE TITLE
Query on library list used while compiling 

### ISSUE THREAD

**Issue by abhishekkalmaki** [halcyon-tech/vscode-ibmi#1281]
Compiling an SQLRPGLE works only when source has no dependency with any external objects but when source has multiple dependency with objects from other libraries and if I add it to library list, compilation fails as below. Also based on IBMi Output, it shows error related to SQL pre-compile errors for missing files which doesn't make sense since the files actually exists in one of the  library list and it would seem like vs code is NOT taking added library list while compiling. Please advise what I am missing.

Error details below.

* QCCSID: 65535
* Features:
   * git: false
   * grep: false
   * tn5250: false
   * setccsid: true
   * md5sum: true
   * bash: true
   * chsh: true
   * GENCMDXML.PGM: true
   * GETNEWLIBL.PGM: true
   * QZDFMDB2.PGM: true
   * startDebugService.sh: true
   * attr: true
   * iconv: true
* SQL enabled: false
* Source dates enabled: false

Variants
```json
{
  "american": "%23@$",
  "local": "%23@$"
}
```

Errors:
```json
[
  {
    "command": "system \"CPYTOSTMF FROMMBR('/QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr') TOSTMF('/tmp/vscodetemp-O_QlPMwJHU') STMFOPT(*REPLACE) STMFCCSID(1208) DBFCCSID(*FILE)\"",
    "code": 255,
    "stderr": "CPFA0A9: Object not found.  Object is /QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr.\nCPFA097: Object not copied.  Object is /QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr.",
    "cwd": "."
  },
  {
    "command": "/QOpenSys/usr/bin/qsh",
    "code": 1,
    "stderr": "liblist: 001-0015 Required operand(s) not specified.",
    "cwd": "/home/user"
  },
  {
    "command": "system \"CPYTOSTMF FROMMBR('/QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr') TOSTMF('/tmp/vscodetemp-O_QlPMwJHU') STMFOPT(*REPLACE) STMFCCSID(1208) DBFCCSID(*FILE)\"",
    "code": 255,
    "stderr": "CPFA0A9: Object not found.  Object is /QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr.\nCPFA097: Object not copied.  Object is /QSYS.lib/TESTLIB.lib/VSCODE.file/RPGLINT.mbr.",
    "cwd": "."
  }
]
```

**Comment by worksofliam** [comment]
We need you to provide a code example and specific steps to recreate this issue. Thanks!

**Comment by worksofliam** [comment]
When you connect to your system, do you get a notice about setting your default shell to bash?

**Comment by abhishekkalmaki** [comment]
Yes, I did got that notice and it's set to bash. Sure, I will recreate that issue and will share you the steps. Thanks!

**Comment by abhishekkalmaki** [comment]
Hi, I have tried to recreate this issue with simple example in my machine and it works fine in that case but when I try with one of the main source code with the exact same step, it doesn't work. Another important point is that, the OUTPUT will show as "Fetching errors from LIB/pgm" and it simply stuck over there. Usually when compilation is done, we will be able to see the spool file output in vs code but in this type of scenarios, it simply stuck at "Fetching errors...". attached screen below. 

Let me know if this helps in finding what might be a setting issue with my side.

![Screenshot_1](https://github.com/halcyon-tech/vscode-ibmi/assets/133192054/7a82a4e3-eb6c-412d-9a92-82774864247e)

**Comment by worksofliam** [comment]
This is an issue because your bash shell is not set correctly. If you're still seeing the warning about bash, set that up.

**Comment by abhishekkalmaki** [comment]
Hi, Sorry for bothering again but I don't see any pop-up or warning about bash. How do I verify this bash setup? or any article that you can share?

**Comment by worksofliam** [comment]
Please share the contents of your Code for IBM i output channel after you try and run the compile and see the `liblist` error.

<img width="939" alt="image" src="https://user-images.githubusercontent.com/3708366/178739532-03c8ad30-84de-4698-9904-65938f2ef651.png">

**Comment by abhishekkalmaki** [comment]
Hi Liam, thanks for pointing me to  "Code for IBM i" output. I have executed the same command directly in IBMi and was getting same liblist error . I did some research and able to figure out the issue that, it's with # character used in library names. Our all application library is using # character at beginning. So I have added the library names within the quotes itself to the library list and now I am able to compile the programs properly. shell will ignore all characters after # and so, that was the cause. Thanks for all your input.
